### PR TITLE
Add more support for fa_IR locale

### PR DESCRIPTION
**Pull request by mousavian**

### CODE REVIEW

File: src/Faker/Provider/fa_IR/Address.php [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        'واتیکان', 'وانواتو', 'ونزوئلا', 'ویتنام',
+        'هائیتی', 'هلند', 'هندوراس', 'هندوستان',
+        'یمن', 'یونان'
+    );
+
+    protected static $state = array(
+        'آذربایجان شرقی', 'آذربایجان غربی', 'اردبیل', 'اصفهان', 'البرز', 'ایلام', 'بوشهر', 'تهران', 'چهارمحال بختیاری', 'خراسان جنوبی', 'خراسان رضوی', 'خراسان شمالی', 'خوزستان', 'زنجان', 'سمنان', 'سیستان و بلوچستان', 'فارس', 'قزوین', 'قم', 'کردستان', 'کرمان', 'کرمانشاه', 'کهکیلویه و بویراحمد', 'گلستان', 'گیلان', 'لرستان', 'مازندران', 'مرکزی', 'هرمزگان', 'همدان', 'یزد'
+    );
+
+    protected static $city = array(
```
Comment (fzaninotto) [inline src/Faker/Provider/fa_IR/Address.php]: What's the datasource? Please read https://github.com/fzaninotto/Faker/blob/master/CONTRIBUTING.md

File: src/Faker/Provider/fa_IR/DateTime.php [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+     *
+     * @param string                 $format
+     * @param \DateTime|int|string   $max    maximum timestamp used as random end limit, default to "now"
+     * @param  boolean $convertChars False (by default) to keep Digits in normal english format, true to convert to Farsi chars(۰,۱,۲,۳,۴,۵,۶,۷,۸,۹)
+     * @return string
+     * @example '1367-09-28'
+     */
+    public static function date($format = 'Y-m-d', $max = 'now', $convertChars = false)
+    {
+        return JalaliDateTime::date($format, static::dateTime($max)->getTimestamp(), $convertChars);
```
Comment (fzaninotto) [inline src/Faker/Provider/fa_IR/DateTime.php]: What's `JalaliDateTime`?

Comment (fzaninotto) [inline src/Faker/Provider/fa_IR/DateTime.php]: Ah, sorry, I just saw it down there. Please add a `use` statement to make the dependency visible.

File: src/Faker/Provider/fa_IR/JalaliDateTime.php [hunk 3]
```diff
@@ -0,0 +1,614 @@
...
+ * List of supported timezones can be found here:
+ * http://www.php.net/manual/en/timezones.php
+ *
+ *
+ * @package    JalaliDateTime
+ * @author     Sallar Kaboli <sallar.kaboli@gmail.com>
+ * @author     Omid Pilevar <omid.pixel@gmail.com>
+ * @copyright  2003-2012 Sallar Kaboli
+ * @license    http://opensource.org/licenses/mit-license.php The MIT License
+ * @link       https://github.com/sallar/jDateTime
```
Comment (fzaninotto) [inline src/Faker/Provider/fa_IR/JalaliDateTime.php]: Don't copy the code, add it as a dependency

File: src/Faker/Provider/fa_IR/Lorem.php [hunk 4]
```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Faker\Provider\fa_IR;
+
+class Lorem extends \Faker\Provider\Lorem
```
Comment (fzaninotto) [inline src/Faker/Provider/fa_IR/Lorem.php]: Lorem is never localized. Please remove this provider.

**Comment by mousavian** [comment]
Thanks for reviewing the code. I've applied mentioned changes.

**Comment by fzaninotto** [comment]
Needs rebase.

**Comment by mousavian** [comment]
@fzaninotto I've rebased this branch. Can you please confirm and merge if everything is okay?

**Comment by fzaninotto** [comment]
Wrong rebase, you've included loads of commits unrelated to this PR. Please try again.

**Comment by fzaninotto** [comment]
Too old PR, closing.

**Pull request closed without merging**